Mathlib/Util/CountHeartbeats.lean

```diff
@@ -117,6 +117,31 @@ elab "count_heartbeats " "in" ppLine cmd:command : command => do
         Lean.Meta.Tactic.TryThis.addSuggestion (← getRef)
           (← set_option hygiene false in `(command| set_option maxHeartbeats $m in $cmd))
 
+/--
+Guard the minimal number of heartbeats used in the enclosed command.
+
+This is most useful in the context of debugging and minimizing an example of a slow declaration.
+By guarding the number of heartbeats used in the slow declaration,
+an error message will be generated if a minimization step makes the slow behaviour go away.
+
+The default number of minimal heartbeats is the value of `maxHeartbeats` (typically 200000).
+Alternatively, you can specify a number of heartbeats to guard against,
+using the syntax `guard_min_heartbeats n in cmd`.
+-/
+elab "guard_min_heartbeats " n:(num)? "in" ppLine cmd:command : command => do
+  let max := (← Command.liftCoreM getMaxHeartbeats) / 1000
+  let n := match n with
+           | some j => j.getNat
+           | none => max
+  let start ← IO.getNumHeartbeats
+  try
+    elabCommand (← `(command| set_option maxHeartbeats 0 in $cmd))
+  finally
+    let finish ← IO.getNumHeartbeats
+    let elapsed := (finish - start) / 1000
+    if elapsed < n then
+      logInfo m!"Used {elapsed} heartbeats, which is less than the guarded quorum of {n}."
```

Comment:
I'm not sure "guarded quorum" is going to be clear to anyone seeing this message. Maybe just "minimum"?